Added example .csv and .html

koenkarsten opened the pull request:
- Added example .csv and .html files as described in https://github.com/mbakholdina/srt-stats-plotting/issues/7.
- Included an info in README on how to use `srt-live-transmit` with the script.

Fixes #7, #8.

koenkarsten committed:
Add example .csv and .html

Adding an example csv and html file as described in https://github.com/mbakholdina/srt-stats-plotting/issues/7.

koenkarsten committed:
Update README.md

koenkarsten committed:
Merge pull request #1 from koenkarsten/master

Update README.md

mbakholdina commented on README.md:
```diff
@@ -20,7 +20,7 @@ pip install -r requirements.txt
 ## Script Usage
 
-The main purpose of `plot_srt_stats.py` script is to visualize all the SRT core statistics produced during experiments by one of the [testing applications](https://github.com/Haivision/srt/blob/master/docs/stransmit.md) or third-party solutions supporting SRT protocol. Depending on whether this statistics is collected on sender or receiver side, the data plotted may vary.
+The main purpose of `plot_srt_stats.py` script is to visualize all the SRT core statistics produced during experiments by one of the [testing applications](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md) or third-party solutions supporting SRT protocol. Depending on whether this statistics is collected on sender or receiver side, the data plotted may vary.
```
Suggested change 1: 
"by one of the [testing applications](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md) or third-party solutions" ->
"by one of the testing applications like [srt-xtransmit](https://github.com/maxsharabayko/srt-xtransmit) and [srt-live-transmit](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md) or third-party solutions"

Suggested change 2:
"to visualize all the SRT core statistics" ->
"to visualize SRT core statistics"

koenkarsten replied:
Resolved in: https://github.com/koenkarsten/srt-stats-plotting/commit/9dca3e38985ff7983eb17c5df3ab6095a165f6f5

mbakholdina commented on README.md:
````diff
@@ -38,6 +38,14 @@ Use `--help` option in order to get the full list of options
 --help        Show this message and exit.
 ```
 
+## Usage with srt-live-transmit
````
I suggest to make the header of level 3 and put it under and at the very end of "Script Usage" section.

koenkarsten replied:
Resolved in https://github.com/koenkarsten/srt-stats-plotting/commit/271e76d50a75d01981afd88951eb9d9b34da9df1

mbakholdina commented on README.md:
````diff
@@ -38,6 +38,14 @@ Use `--help` option in order to get the full list of options
 ```
 
+## Usage with srt-live-transmit
+As mentioned this tool is intended for usage with [srt-live-transmit](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md), for which you explicitly need to enable the export of statistics in .csv format. This can be done by passing the following parameters:
````
Not exactly, another test applications like srt-xtransmit (I refer to it a bit earlier) or any other application producing csv files with the same column names can be used as a generator of SRT statistics for the script. So let's formulate this in a bit different way.

"As mentioned this tool is intended for usage with [srt-live-transmit](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md), for which you explicitly need to enable the export of statistics in .csv format." -> "For the usage with [srt-live-transmit](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md), make sure that the export of SRT statistics is enabled in `.csv` format. This can be done ..."

koenkarsten replied:
Resolved in https://github.com/koenkarsten/srt-stats-plotting/commit/f2c97e0c19c8efa5cc1b9bc5b4721879d260376a

mbakholdina commented on README.md:
```diff
@@ -38,6 +38,14 @@ Use `--help` option in order to get the full list of options
+As mentioned this tool is intended for usage with [srt-live-transmit](https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md), for which you explicitly need to enable the export of statistics in .csv format. This can be done by passing the following parameters:
+- ` -statspf:csv` to ensure the CSV format is used when exporting.
+- ` -statsout:/path/to/desired/folder/filename-rcv.csv` to define where to write the statistics
+- ` -stats-report-frequency:1000` to define how often to write statistics (in ms), this example sets it to once per second.
```
Not exactly. In `srt-live-transmit` the frequency is measured in the number of packets. See the description of the option "The frequency of SRT statistics collection, based on the number of packets." If you specify 1000, it means the statistics is calculated each 1000 packets.

To collect the statistics each N ms, please transition to [srt-xtransmit](https://github.com/maxsharabayko/srt-xtransmit) which is more powerful and handy for data analysis I believe. You will find the examples in README. There are 2 options for this `--statsfile stats-snd.csv --statsfreq 100ms`.

If you have time and interest, you can add the bit about `srt-xtransmit` here as well.

koenkarsten replied:
Resolved in https://github.com/koenkarsten/srt-stats-plotting/commit/179fc9b02a77a96af7d18433b05490f72f8f7d57

mbakholdina commented on README.md:
```diff
@@ -49,7 +57,7 @@ where _interval_ is the interval indicating how frequently the statistics is col
 ## Example Plots
 
-The script plots different charts from SRT statistics. For example, a chart to visualize statistics on the packets being sent, lost, retransmitted, dropped or on flight:
+The script plots different charts from SRT statistics, an example .csv file can be found in the examples folder, together with the corresponding output html. . For example, a chart to visualize statistics on the packets being sent, lost, retransmitted, dropped or on flight:
```
Suggested change:
"The script plots different charts from SRT statistics, an example .csv file can be found in the examples folder, together with the corresponding output html. ." -> 
"The script plots different charts from SRT statistics. Examples of `.csv` files and corresponding `.html` output can be found in the [examples](./examples) folder."

mbakholdina replied:
I would suggest to create a folder "srt-live-transmit" under "examples" folder and put your files in there. It would be also helpful if you could generate sender side statistics and add examples for it as well.

koenkarsten replied:
resolved in https://github.com/koenkarsten/srt-stats-plotting/commit/718676727c4f42878ea9030a123704e6ee2efe63

mbakholdina reviewed (changes requested):
Hi @koenkarsten,
Thank you for your PR, it's super helpful!

Please find my comments and suggested changes here. Let me know in case you have any further questions.

Thanks,
Maria

koenkarsten committed:
Resolve https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787385652

https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787385652

koenkarsten committed:
Resolve https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787382041

https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787382041

koenkarsten committed:
Resolve https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787382041

https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787382041

koenkarsten committed:
Resolve https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787375884

https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787375884

koenkarsten committed:
Resolve https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787374628

https://github.com/mbakholdina/srt-stats-plotting/pull/10/files/b6db4fa4bc3c7449ecc018262e0bf695033ca63e#r787374628

koenkarsten commented:
Hey @mbakholdina, thanks for checking out the PR! I've had a look and addressed your comments, could you give it another look? Thanks! Cheers, Koen.

mbakholdina commented:
Hi @koenkarsten,
Thank you for improving the PR, merged!
